File: modules/cubex-i18n/src/main/java/org/cubexmc/i18n/Placeholders.java
package org.cubexmc.i18n;

import java.util.HashMap;
import java.util.Map;

public final class Placeholders {

    private Placeholders() {
    }

    public static Map<String, Object> empty() {
        return new HashMap<>();
    }

    public static Map<String, Object> of(String key, Object value) {
        Map<String, Object> placeholders = empty();
        placeholders.put(key, value);
        return placeholders;
    }

    public static Map<String, Object> put(Map<String, Object> args, String key, Object value) {
        args.put(key, value);
        return args;
    }
}

File: modules/cubex-i18n/src/main/kotlin/org/cubexmc/i18n/Placeholders.kt
package org.cubexmc.i18n

object Placeholders {
    @JvmStatic
    fun empty(): MutableMap<String, Any?> = HashMap()

    @JvmStatic
    fun of(key: String, value: Any?): MutableMap<String, Any?> = empty().also { it[key] = value }

    @JvmStatic
    fun put(args: MutableMap<String, Any?>, key: String, value: Any?): MutableMap<String, Any?> =
        args.also { it[key] = value }
}
